### DISCUSSION TITLE
Acetyl lysine (ALY) fails to be recognized by pdb2pqr

### DISCUSSION THREAD

**Discussion by Aekith** [Electrostatics/pdb2pqr#395] [Q&A]
Hello. I have a pdb file that I acetylated at the level of a Lys residue, and pdb2pqr rightly fails to recognize it as a canonical amino acid and skips it altogether. Any suggestions on how to make pdb2pqr recognize my file? I'm using CHARMM36 as the force field and would like to then review it with a specific topology file that can recognize Aly, which is "toppar_all36_prot_modify_res.str". I used the following command:

`pdb2pqr --ff=charmm --ffout=charmm inputfilename.pdb outputfilename.pqr`

**Comment by sobolevnrm** [comment]
Hello.  You will need to extend/modify the existing PDB2PQR force field.  Some documentation on this is available at https://pdb2pqr.readthedocs.io/en/latest/extending.html#adding-new-forcefield-parameters.

**Reply by Aekith** [reply]
Thank you! I tried to follow the guide, but I still can't generate a pqr file that indicates an error from the PDB2PQR server. For this reason, and I apologize in advance if this is a basic question, but it's been a few months since I started delving into this, I found a CHARMM.dat file in the repositories that is inspired by the CHARMM22 force field. I am working with CHARMM36, and I cannot find a pre-existing .dat file for this specific version. Since the values of ALY in the file I found are specific to the CHARMM36 force field, does this mean that if I extend the .dat file inspired by version 22, it would create conflicts?






Because when I tried to extend the existing file CHARMM.dat with values of ALY from 'toppar_all36_prot_modify_res.str'  run it from PDB2PQR server it gaves me this warning and error from the website:

`2024-05-04 17:04:22,994 WARNING:biomolecule.py:741:apply_force_field:Residue ALY A 138 has non-integer charge: -1.7 deviates by 0.30000000000000004 from integral, exceeding error tolerance 0.001. 
2024-05-04 17:04:22,995 WARNING:main.py:699:non_trivial:Residue ALY A 138 has non-integer charge:  -1.7 deviates by 0.30000000000000004 from integral, exceeding error tolerance 0.001
2024-05-04 17:04:22,995 CRITICAL:main.py:800:main_driver:2.3 deviates by 0.2999999999999998 from integral, exceeding error tolerance 0.001
2024-05-04 17:04:22,995 CRITICAL:main.py:801:main_driver:Giving up.

`

Here attacched I put a zip file of the files that I modified and "adapted" (myff.dat, my.names and also I included the 'toppar_all36_prot_modify_res.str' from where I took the values of ALY).
[ALY attempt.zip](https://github.com/Electrostatics/pdb2pqr/files/15212218/ALY.attempt.zip)

**Reply by sobolevnrm** [reply]
Have you added up the charges for the atoms in your original parameter file and confirmed that they sum to an integer?
